client/signup/steps/difm-site-picker/index.tsx

```diff
@@ -0,0 +1,169 @@
+	);
+};
+
+export default function DIFMSitePickerStep( props: Props ): React.ReactElement {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	const { goToNextStep } = props;
+	const [ siteId, setSiteId ] = useState< number | null >( null );
+	const [ confirmDomain, setConfirmDomain ] = useState( '' );
+	const siteDomain = useSelector( ( state ) => getSiteDomain( state, siteId ) );
+	const siteSlug = useSelector( ( state ) => getSiteSlug( state, siteId ) );
+	const siteTitle = useSelector( ( state ) => getSiteTitle( state, siteId ) );
+	const headerText = translate( 'Choose where you want us to build your site.' );
+	const subHeaderText = translate( 'Some unsupported sites may be hidden.' );
+
+	useEffect( () => {
+		dispatch( saveSignupStep( { stepName: props.stepName } ) );
+	}, [ dispatch, props.stepName ] );
+
+	const handleSiteSelect = ( siteId: number ) => {
+		setSiteId( siteId );
+	};
+
+	const filterSites = ( site: SiteData ) => {
+		return (
+			site.capabilities?.manage_options &&
+			! site.jetpack &&
+			! site.options?.is_wpforteams_site &&
+			! site.options?.is_difm_lite_in_progress
+		);
+	};
+
+	const onCloseDialog = () => {
+		setSiteId( null );
+	};
+
+	const onConfirmDelete = () => {
+		dispatch(
+			submitSignupStep(
+				{
+					stepName: props.stepName,
+					stepSectionName: props.stepSectionName,
+					siteId,
+					siteSlug,
+				},
+				{
+					siteId,
+					siteSlug,
+				}
+			)
+		);
+		//Skip domains step
+		dispatch(
+			submitSignupStep(
+				{
+					stepName: 'domains',
+					wasSkipped: true,
+				},
+				{
+					domainItem: undefined,
+					themeItem: undefined,
+				}
+			)
+		);
+
+		goToNextStep();
+	};
+
+	if ( siteId ) {
+		const deleteDisabled = false;
+		// typeof confirmDomain !== 'string' ||
+		// confirmDomain.toLowerCase().replace( /\s/g, '' ) !== siteDomain;
+		const buttons = [
+			<Button onClick={ onCloseDialog }>{ translate( 'Cancel' ) }</Button>,
+			<Button primary scary disabled={ deleteDisabled } onClick={ onConfirmDelete }>
+				{ translate( 'Delete this site' ) }
+			</Button>,
+		];
+
+		return (
+			<Dialog isVisible={ true } buttons={ buttons } onClose={ onCloseDialog }>
+				<h1>{ translate( 'Confirm delete content' ) }</h1>
+				<p>
+					{ translate(
+						'The contents of your site {{strong}}%(siteTitle)s{{/strong}} (%(siteAddress)s) will be permanently deleted. ' +
+							'This includes all posts, pages, media, and comments. ' +
+							'Once you complete the purchase, your site will remain inaccessible while we rebuild your site.',
+						{
+							components: {
+								strong: <strong />,
+							},
+							args: {
+								siteTitle,
+								siteAddress: siteDomain,
+							},
+						}
+					) }
+				</p>
+				<p>{ translate( 'The content will be deleted only after the purchase.' ) }</p>
+				<FormLabel htmlFor="confirmDomainChangeInput">
+					{ translate(
+						'Please type in {{warn}}%(siteAddress)s{{/warn}} in the field below to confirm. ' +
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **24** times:
`translate( 'Please type in {{warn}}%(siteAddress)s{{/warn}} in the field below to confirm. Your site will then be gone forever.' )` *ES Score: 9*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/new-or-existing-step-in-difm)